packages/autocomplete-theme-classic/src/theme.scss

```diff
@@ -1,351 +1,451 @@
+    sans-serif;
+  // icons
   --aa-icon-size: 18px;
-  --aa-icon-stroke-width: calc((20 / var(--aa-base-unit)) * 1.6);
-  --aa-primary-color: rgb(62, 52, 211);
-  --aa-muted-color: rgba(128, 126, 163, 0.6);
-  --aa-selected-color: rgba(62, 52, 211, 0.1);
-  --aa-icon-color: rgb(119, 119, 163);
-  --aa-text-color: rgb(38, 38, 39);
-  --aa-content-text-color: rgb(38, 38, 39, 0.7);
-  --aa-background-color: rgb(255, 255, 255);
-  --aa-background-color-alpha-0: rgba(255, 255, 255, 0);
+  --aa-icon-stroke-width: 1.6;
+  --aa-icon-color-rgb: 119, 119, 163;
+  --aa-icon-color-alpha: 1;
+  // text colors
+  --aa-text-color-rgb: 38, 38, 39;
+  --aa-text-color-alpha: 1;
+  --aa-primary-color-rgb: 62, 52, 211;
+  --aa-primary-color-alpha: 0.2;
+  --aa-muted-color-rgb: 128, 126, 163;
+  --aa-muted-color-alpha: 0.6;
+  // border colors
+  --aa-panel-border-color-rgb: 128, 126, 163;
+  --aa-panel-border-color-alpha: 0.3;
+  --aa-input-border-color-rgb: 128, 126, 163;
+  --aa-input-border-color-alpha: 0.8;
+  // background colors
+  --aa-selected-color-rgb: 179, 173, 214;
+  --aa-selected-color-alpha: 0.205;
+  --aa-background-color-rgb: 255, 255, 255;
+  --aa-background-color-alpha: 1;
+  --aa-input-background-color-rgb: 255, 255, 255;
+  --aa-input-background-color-alpha: 1;
+  // detached mode
+  --aa-detached-media-query: (max-width: 680px);
+  --aa-detached-modal-media-query: (min-width: 680px);
+  --aa-detached-modal-max-width: 680px;
+  --aa-detached-modal-max-height: 500px;
+  --aa-overlay-color-rgb: 115, 114, 129;
+  --aa-overlay-color-alpha: 0.4;
+  // shadows
   --aa-panel-shadow: 0 0 0 1px rgba(35, 38, 59, 0.1),
     0 6px 16px -4px rgba(35, 38, 59, 0.15);
-  --aa-panel-max-height: 500px;
-  --aa-detached-media-query: (max-width: 500px);
-  --aa-detached-modal-media-query: (min-width: 500px);
-  --aa-detached-modal-max-width: 500px;
-  --aa-detached-modal-max-height: 500px;
+  --aa-key-shadow: '';
 }
 
 // ----------------
-// Darkmode
+// 2. Dark Mode
 // ----------------
 body {
   /* stylelint-disable selector-no-qualifying-type, selector-class-pattern */
   &[data-theme='dark'],
   &.dark {
-    --aa-primary-color: rgb(93, 85, 213);
-    --aa-muted-color: rgba(93, 85, 213, 0.6);
-    --aa-selected-color: rgba(93, 85, 213, 0.25);
-    --aa-icon-color: rgb(119, 119, 163);
-    --aa-text-color: rgb(183, 192, 199);
-    --aa-content-text-color: rgb(183, 192, 199, 0.8);
-    --aa-background-color: rgb(21, 24, 42);
+    --aa-primary-color-rgb: 93, 85, 213;
+    --aa-muted-color-rgb: 93, 85, 213;
+    --aa-selected-color-rgb: 93, 85, 213;
+    --aa-selected-color-alpha: 0.25;
+    --aa-icon-color-rgb-rgb: 119, 119, 163;
+    --aa-text-color-rgb: 183, 192, 199;
+    --aa-background-color-rgb: 21, 24, 42;
     --aa-panel-shadow: inset 1px 1px 0 0 rgb(44, 46, 64),
       0 3px 8px 0 rgb(0, 3, 9);
   }
   /* stylelint-enable selector-no-qualifying-type, selector-class-pattern */
 }
 
+// Reset for `@extend`
+%reset {
```

Comment:
What was the rationale behind using placeholder (to use with `@extend`) rather than mixins? File size?

My usual concern with `@extend` is that you don't control the final source order, so you can easily run into issues where the [final style isn't what you expect](https://codepen.io/sarahdayanalgolia/pen/poRvPrx). Also, the final size might not be as optimized because it generates long comma-separated selectors that can end up being a lot heavier and not compress as well as a few duplicated rules.

The `@extend` directive is still useful if used well but it can easily bite you, so unless we've already carefully considered it, maybe we want to move to a mixin instead. We can always compare both versions after gzip/Brotli-ing them if we're concerned about file size.